src/main/java/net/minecraftforge/client/gui/config/ListConfigScreen.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2020.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.gui.config;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.widget.button.Button;
+import net.minecraft.util.text.IFormattableTextComponent;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraftforge.client.gui.config.ControlCreator.Interactor;
+import net.minecraftforge.common.util.NonNullSupplier;
+import net.minecraftforge.fml.client.gui.widget.ExtendedButton;
+import org.apache.commons.lang3.mutable.MutableObject;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.BooleanSupplier;
+
+/**
+ * Popup {@link ConfigScreen} that displays and modifies a {@link List}.
+ *
+ * @author Cadiboo
+ */
+public abstract class ListConfigScreen extends ConfigScreen {
+    protected List value;
+
+    public ListConfigScreen(Screen parentScreen, ITextComponent title, List<?> value) {
+        super(parentScreen, title, ConfigScreen.makeSubtitle(parentScreen, title));
+        this.value = value;
+    }
+
+    public abstract void onModified(List newValue);
+
+    @Override
+    protected ConfigElementList makeConfigElementList() {
+        return new ListConfigElementList(this, field_230706_i_);
+    }
+
+    protected void setAndNotify(int index, Object newValue) {
+        value.set(index, newValue);
+        onModified(value);
+    }
+
+    protected void onAddRemove(int index) {
+        // TODO: Don't take the lazy way out
```

Comment:
Shouldn't matter that much, doesn't seem like a performance hot spot

Reply:
It’s not, the issue is that whenever someone adds/removes an element the screen refreshes from scratch, losing the “undo” ability for any changes they’d made before adding/removing.
I plan to generify this class so it can be a superclass for both List screens and Config screens (raw ones from NightConfig, effectively `Map<String, ?>`s).